### PR TITLE
update logic for worker removal in check ttl

### PR DESCRIPTION
**Pull request by quasiben**
PR updates #3896 

cc @jonashaag

### CODE REVIEW

File: distributed/scheduler.py [hunk 1]
```diff
@@ -5304,8 +5304,11 @@ async def check_worker_ttl(self):
         now = time()
         for ws in self.workers.values():
             if (ws.last_seen < now - self.worker_ttl) and (
-                10 * heartbeat_interval(len(self.workers))
+                ws.last_seen < now - 10 * heartbeat_interval(len(self.workers))
             ):
+                print(
+                    now, ws.last_seen, now - 10 * heartbeat_interval(len(self.workers))
+                )
```
Comment (mrocklin) [inline distributed/scheduler.py:5311]:
```suggestion
```

File: distributed/tests/test_failed_workers.py [hunk 2]
```diff
@@ -417,14 +417,19 @@ async def test_restart_timeout_on_long_running_task(c, s, a):
...
+
     a.periodic_callbacks["heartbeat"].stop()
     await asyncio.sleep(0.010)
     assert set(s.workers) == {a.address, b.address}
 
     start = time()
     while set(s.workers) == {a.address, b.address}:
-        await asyncio.sleep(0.050)
-        assert time() < start + 2
+        await asyncio.sleep(interval)
```
Comment (mrocklin) [inline distributed/tests/test_failed_workers.py:432]: How long does this test take to run?  If it's longer than a second can we mark it with `pytest.mark.slow`.  If it's longer than 10 seconds then we might want to find some other way around this.

**Comment by mrocklin** [comment]
Thanks for doing this @quasiben !

File: distributed/scheduler.py (hunk 1 above)
Comment (quasiben) [reply distributed/scheduler.py:5311]: Oops.  Thanks!

File: distributed/tests/test_failed_workers.py (hunk 2 above)
Comment (quasiben) [reply distributed/tests/test_failed_workers.py:432]: This takes ~7 seconds. 
> real    0m7.363s

Happy to make as slow if that's close enough to 10

**Comment by mrocklin** [comment]
Let's definitely mark it as slow.

**Pull request merged**